warp10/src/main/java/io/warp10/continuum/gts/GTSPickler.java

```diff
@@ -0,0 +1,102 @@
+//
+//   Copyright 2019  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.continuum.gts;
+
+import com.geoxp.GeoXPLib;
+import net.razorvine.pickle.IObjectPickler;
+import net.razorvine.pickle.PickleException;
+import net.razorvine.pickle.Pickler;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Pickler for Geo Time Series
+ */
+public class GTSPickler implements IObjectPickler {
+
+  private static String CLASSNAME_KEY = "classname";
+  private static String LABELS_KEY = "labels";
+  private static String ATTRIBUTES_KEY = "attributes";
+  private static String TIMESTAMPS_KEY = "timestamp";
+  private static String VALUES_KEY = "value";
+  private static String LATITUDE_KEY = "latitude";
+  private static String LONGITUDE_KEY = "longitude";
+  private static String ELEVATION_KEY = "elevation";
+
+  public void pickle(Object o, OutputStream out, Pickler currentPickler) throws PickleException, IOException {
+    GeoTimeSerie gts = (GeoTimeSerie) o;
+
+    Map<String, Object> gtsAsMap = new HashMap<>();
+    gtsAsMap.put(CLASSNAME_KEY, gts.getName());
+    gtsAsMap.put(LABELS_KEY, gts.getMetadata().getLabels());
+    gtsAsMap.put(ATTRIBUTES_KEY, gts.getMetadata().getAttributes());
+
+    List<Long> ticks = new ArrayList<Long>(gts.values);
+    for (int i = 0; i < gts.values; i++) {
+      ticks.add(gts.ticks[i]);
+    }
+    gtsAsMap.put(TIMESTAMPS_KEY, ticks);
+
+    if (0 == gts.values) {
+      gtsAsMap.put(VALUES_KEY, new long[0]);
+    } else {
+      List<Object> values = new ArrayList<Object>(gts.values);
+
+      for (int i = 0; i < gts.values; i++) {
+        values.add(GTSHelper.valueAtIndex(gts, i));
+      }
+      gtsAsMap.put(VALUES_KEY, values);
+    }
+
+    if (gts.hasLocations()) {
+      long[] locations = gts.locations;
+      List<Double> lats = new ArrayList<Double>();
```

Comment:
If pickle does not care if those are floats or doubles, make that floats as it will be less resource hungry.

Reply:
Python does care about double precision (in fact what it calls float have double precision and it doesn't call anything double)

Reply:
The question is, does pickle actually convert Float to doubles? If so you can allocate Floats instead of Doubles since lat/lon are really stored on 32bits each

Reply:
yes that's the case, it converts java floats to python floats (which are the same precision than java doubles)